modules/puppet_connect/plans/test_input_data.pp

```diff
@@ -0,0 +1,31 @@
+# @summary
+#   Tests that the provided Puppet Connect input data is complete, meaning that all consuming inventory targets are connectable.
+#
+# This plan should only be used as part of the copy-pastable "test input data"
+# workflow specified in the Puppet Connect docs.
+#
+# @param targets
+#   The inventory targets. This should never be set by the user. It is only here for compatibility with Bolt's rerun workflow.
+#
+# @return ResultSet the result of invoking the 'is connectable?' query on all
+# the targets. Note that this query currently consists of running the 'whoami'
+# command.
+#
+plan puppet_connect::test_input_data(TargetSpec $targets = 'all') {
+  $targs = get_targets($targets)
+  $targs.each |$target| {
+    if $target.transport != 'ssh' and $target.transport != 'winrm' {
+      fail_plan("Inventory contains target ${target} with unsupported transport, must be ssh or winrm")
+    }
+    if $target.host == undef {
```

Comment:
I _think_ this check should be unnecessary since the SSH/WinRM transports will report that the target doesn't have a host. Having a `fail_plan` here means you will only see the error for the _first_ target that's missing a host.

Reply:
Yup, you are correct. Updated.